### ISSUE TITLE
KotlinGenerator and ScalaGenerator don't produce correct POJO.toString(), equals(), hashCode() methods for array types

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#10787]
When tables containing array types are generated, the generated `toString()` method doesn't produce the correct content for those array types. E.g. after fixing https://github.com/jOOQ/jOOQ/issues/10768

This table (using HSQLDB syntax):

```sql
CREATE TABLE arrays.t (
  id int,
  
  i int ARRAY,
  l bigint ARRAY,
  s varchar(10) ARRAY,
  d date ARRAY,
  
  CONSTRAINT pk_t PRIMARY KEY (id)
)
```

Produces this kotlin code

```kotlin
class T(
      var id: Int? = null
    , var i: Array<Int?>? = null
    , var l: Array<Long?>? = null
    , var s: Array<String?>? = null
    , var d: Array<Date?>? = null
): Serializable {


    override fun toString(): String {
        val sb = StringBuilder("T (")

        sb.append(id)
        sb.append(", ").append(i)
        sb.append(", ").append(l)
        sb.append(", ").append(s)
        sb.append(", ").append(d)

        sb.append(")")
        return sb.toString()
    }
}
```

The corresponding Java code produces `append(Arrays.toString(i))` calls, and similar.

**Comment by lukaseder** [comment]
This also applies to the `ScalaGenerator`

**Comment by lukaseder** [comment]
Perhaps, `equals()` and `hashCode()` are also incorrect...?

**Comment by lukaseder** [comment]
Yep, also incorrect

**Comment by lukaseder** [comment]
This isn't so severe, probably, now that people will generate Kotlin data classes, or Scala case classes. But maybe, they won't do that.

**Comment by lukaseder** [comment]
Scala array equality:

```scala
a sameElements b
```

Hashcode

```scala
a.toSeq.hashCode
```

**Comment by lukaseder** [comment]
Fixed for jOOQ 3.15.0. Backport scheduled for 3.14.1 (#10790)
